SEP-0012.md

```diff
@@ -0,0 +1,537 @@
+    @property
+    def combined_wcs(self) -> BaseHighLevelWCS:
+        """
+        The WCS transform for the NDCube, including the coordinates specified in ``.extra_coords``.
+
+        This transform should implement the high level wcsapi, and have
+        `pixel_n_dim` equal to the number of array dimensions in the
+        `.NDCube`. The number of world dimensions should be equal to the
+        number of world dimensions in ``self.wcs`` and in ``self.extra_coords`` combined.
+        """
+
+    @abc.abstractmethod
+    @property
+    def array_axis_physical_types(self) -> Iterable[Tuple[str, ...]]:
+        """
+        Returns the WCS physical types associated with each array axis.
+
+        Returns an iterable of tuples where each tuple corresponds to an array axis and
+        holds strings denoting the WCS physical types associated with that array axis.
+        Since multiple physical types can be associated with one array axis, tuples can
+        be of different lengths. Likewise, as a single physical type can correspond to
+        multiple array axes, the same physical type string can appear in multiple tuples.
+        """
+
+    @abc.abstractmethod
+    def axis_world_coords(self,
+                          *axes: Union[int, str],
+                          edges: bool = False,
+                          wcs: Optional[Union[BaseHighLevelWCS, ExtraCoordsABC]] = None
+                          ) -> Iterable[Any]:
+        """
+        Returns objects representing the world coordinates of all pixel centers.
+
+        Parameters
+        ----------
+        axes
+            Axis number(s) in numpy ordering or unique substring of
+            `.NDCube.wcs.world_axis_physical_types` or
+            `.NDCube.wcs.world_axis_names` of axes for which real world
+            coordinates are desired. Not specifiying axes implies all axes will be
+            returned.
+
+        edges
+            If `True` then instead of returning the coordinates of the pixel
+            centers the coordinates of the pixel edges will be returned.
+
+        wcs
+            The WCS object to used to calculate the world coordinates.
+            Although technically this can be any valid WCS, it will typically be
+            ``self.wcs``, ``self.extra_coords``, or ``self.combined_wcs``, combing both
+            the WCS and extra coords.
+            Default=self.wcs
+
+        Returns
+        -------
+        axes_coords
+            High level object, or iterable thereof, giving the real world
+            coords for the axes requested by user.
+            For example, SkyCoords. The types returned are determined by the WCS object.
+            The dimensionality of these objects should match that of their corresponding array dimensions.
+
+        Example
+        -------
+        >>> NDCube.all_world_coords(('lat', 'lon'))
+        >>> NDCube.all_world_coords(2)
+
+        """
+
+    @abc.abstractmethod
+    def axis_world_coords_values(self,
```

Comment:
In theory yes, but that mapping isn't always clear. For complex WCSes the mapping back from the objects returned to the array coordinates isn't always obvious and it's hard to predict how to extract the values from the objects. Much like why this split is in APE 14 (where the nomenclature comes from) for a reason it is duplicated here because some times you want one and sometimes you want the other.